pkg/csi/nodeserver.go

```diff
@@ -463,7 +475,73 @@ func (ns *nodeServer) resizeVolume(ctx context.Context, volumeID, targetPath str
 	return nil
 }
 
-func (ns *nodeServer) setIOThrottling(ctx context.Context, req *csi.NodePublishVolumeRequest) (err error) {
+// Detect cgroup version by using stat cmd
+func (ns *nodeServer) detectCgroupVersion() string {
+	cgroupVersion := utils.CgroupV1
+	// https://kubernetes.io/docs/concepts/architecture/cgroups/#check-cgroup-version
+	cmd := fmt.Sprintf("stat -fc %%T %s/fs/cgroup", ns.options.sysPath)
+	output, err := ns.osTool.RunShellCommand(cmd)
+	if err != nil {
+		log.Errorf("detectCgroupVersion exec cmd '%s' failed: %s, output: %s, falling back to use %s",
+			cmd, err.Error(), output, cgroupVersion)
+		return cgroupVersion
+	}
+	// For cgroup v2, the output is cgroup2fs.
+	// For cgroup v1, the output is tmpfs.
+	if strings.Contains(output, "tmpfs") {
+		cgroupVersion = utils.CgroupV1
+	} else if strings.Contains(output, "cgroup2fs") {
+		cgroupVersion = utils.CgroupV2
+	} else {
+		log.Errorf("detectCgroupVersion unrecognized device type: %s, falling back to use %s",
+			output, cgroupVersion)
+	}
+	return cgroupVersion
+}
+
+// Detect cgroup driver by searching the group path of the current Pod
+// in the cgroup directory tree
+func (ns *nodeServer) detectCgroupDriver() utils.CgroupDriverType {
```

Comment:
A so delicate way.

Reply:
The word "delicate" has a lot of meanings when translated into Chinese, if you mean it's fragile, then I have a few comments :)